src/ga/ga_selectors.rs

```diff
@@ -9,13 +9,11 @@ use std::cmp;
 /// Interface to Selection Schemes
 pub trait GASelector<'a, T: GASolution>
 {
-    /// Assign the population on which to operate
-    fn assign(&mut self, population: &'a mut GAPopulation<T>);
     /// Update internal state. 
     /// Some selectors implement an empty update().
-    fn update(&mut self) {}
+    fn update(&mut self, population : &mut GAPopulation<T>) {}
 
-    fn select(&self) -> &T;
+    fn select(&self, population: &'a GAPopulation<T>) -> &'a T;
```

Comment:
Select shouldn't require mutability, since we expect several selects to happen:
parent1 = selector.select(...);
parent2 = selector.select(...);

Update can have a mut reference, since we expect update to change it and the borrow ends immediately on return (unlike select, that returns a reference into the population)